lib/tasks/email_missed_reservations.rake

```diff
@@ -0,0 +1,16 @@
+desc 'Send emails about missed reservations'
+task email_missed_reservations: :environment do
+  # get all reservations that are approved and missed, send an email to
+  # user to inform them
+  missed_reservations = Reservation.where(
+    "checked_out IS NULL and approval_status <> 'denied' and due_date < ? ",
```

Comment:
I think we want to exclude requests that were both denied (i.e. `approval_status = 'denied'`) and those that simply weren't processed (i.e. `approval_status = 'requested'`). That means that we'll only be including those with an  approval status of `auto` or `approved`. You can structure the query either way, I'd say it's probably better to use the equality since that way we don't unexpectedly send e-mails to people if we introduce a different `approval_status` option (e.g. `pending`).